bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/type/IChannelTypeBuilder.java

```diff
@@ -24,7 +25,7 @@
  *
  */
 @NonNullByDefault
-interface IChannelTypeBuilder<T extends IChannelTypeBuilder<T>> {
+public interface IChannelTypeBuilder<T extends IChannelTypeBuilder<T>> {
```

Comment:
Why do we use a "I" in fron of the "ChannelTypeBuilder". I was not aware that we mark interface by a naming scheme (and I don't see any reason for that).

Reply:
The reason is simply that the name `ChannelTypeBuilder` is taken by the builder itself... if you have another name for the interface, I am open for suggestions :)

Reply:
Why was this IChannelTypeBuilder interface moved to an internal package @triller-telekom? 
It looks like I can't use any of the methods on this interface now from other bundles. E.g. `build()`.
This makes `org.eclipse.smarthome.core.thing.type.ChannelTypeBuilder` unusable. 
Or am I missing something here? :-)

Reply:
@wborn You are right

https://github.com/triller-telekom/smarthome/blob/e8c71fb00814ddba3cd497b8caa3a4cc764d233f/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/StateChannelTypeBuilder.java

is a public interface (as intended) but it extends an internal interface...
This is a violation and should be detected as soon as https://github.com/openhab/static-code-analysis/issues/210 is integrated.

@triller-telekom Please fix...

About the name:
The `ChannelTypeBuilder` seems to me more like a `ChannelTypeBuilderFactory` and the `IChannelTypeBuilder` as the `ChannelTypeBuilder` itself.

Reply:
Yes the name is confusing also because it just differs one letter. 
You could also add a prefix like: General, Common, Shared, Generic
That is because it provides the methods that are common to all channel type builders.

Reply:
I would prefer to have the `IChannelTypeBuilder` interface be renamed to something @wborn mentioned. Otherwise the usage would be a little bit clunky and unintuitive when a `ChannelTypeBuilderFactory` is the entry point.